### DISCUSSION TITLE
Switch OCIO mode back to Legacy?

### DISCUSSION THREAD

**Discussion by herronelou** [AcademySoftwareFoundation/OpenRV#538] [Q&A] [answered]
Hello.

We recently started using RV 2024.1.0, which is based on OpenRV. 
The new full OCIO implementation is great to have, however a good portion of our users have reported performance issues when using the newer version of RV.

We noticed that playback performance increases again when disabling OCIO, and varies depending on the complexity of the OCIO colorspaces chosen.

I was hoping there might an environment variable or something like that to put back RV in old "Make a LUT our of OCIO" mode to see if that makes a difference with our performance. Is there any such thing?

**Comment by geffrak** [comment]
There seems to be a way and it's explained in the pull request that implemented the GPU Processor (https://github.com/AcademySoftwareFoundation/OpenRV/pull/394).

`export RV_OCIO_USE_LEGACY_GPU_PROCESSOR=1`

Is it what you are looking for?

**Reply by herronelou** [reply]
That does look like what I'm looking for, yes. Somehow I didn't find that through a GitHub search in code. I'm going to give it a try and see if that brings back our playback performance.

**Reply by herronelou** [reply]
I don't know if the env-var is working or not, I'm not sure if there's a way to tell, all I can tell is that OCIO is affecting my playback now in ways it wasn't before. I guess I should try RV 2024.0 as that predates the new gpu processor.

**Reply by bernie-laberge** [reply]
Hello @herronelou , would it be possible to provide us with some repro steps of this OCIOv2 performance issue so that it can get investigated ?
Thanks!

**Reply by herronelou** [reply]
Good timing @bernie-laberge I sent 2 recordings to Gabrielle over on the Shotgrid support side just yesterday.
Repro is going to be difficult, even on our side it seems to be affected by external factors that I haven't been able to identify, for example I was able to reproduce on my machine for days and then it suddenly fixed itself and I haven't been able to reproduce since, while it still affected other users. Then it would stop affecting some of these users and affect others. I've been trying to get it on screen recordings since late July (I've mentioned the issue to Eli at Siggraph) but it's been evading capture because the artists / supes complaining about it were never available to show me while it was happening.

**Reply by herronelou** [reply]
Mini update as I see this again, the playback issues also happened in 2024.0 so may not be related to the new OCIO processing, although it is most definitely related to something with OCIO.
Installing 2024.2 to see if things are any better, if not we'll find another player.

**Reply by herronelou** [reply, marked as answer]
https://github.com/AcademySoftwareFoundation/OpenRV/commit/59cee81d90cf810ab9ec5ddc83d3b64370718093 has fixed the issues we were seeing
